hw/ip/rv_core_ibex/data/rv_core_ibex_testplan.hjson

```diff
@@ -0,0 +1,327 @@
+              - after exiting debug mode, core will execute one instruction and re-enter
+                debug mode'''
+      milestone: V1
+      tests: ["debug"]
+    }
+    {
+      name: interrupt_test
+      desc: '''
+            Assert debug_req_i during CSR modification operations, to force core
+            into debug mode.'''
+      milestone: V1
+      test: ["irq"]
+    }
+    {
+      name: interrupt_wfi_test
+      desc: '''
+            Randomly assert interrupt lines while the core is sleeping due to a WFI instruction'''
+      milestone: V1
+      test: ["irq"]
+    }
+    {
+      name: interrupt_csr_test
+      desc: '''
+            Randomly assert interrupt lines during CSR modification operations,
+            core should jump to proper handler'''
+      milestone: V1
+      test: ["irq"]
+    }
+    {
+      name: interrupt_nested_test
+      desc: '''
+            - randomly assert interrupt lines during program execution
+            - verify that jumps to interrupt handler
+            - trigger another interrupt during first intr_handler execution
+            - verify that core takes this nested interrupt and handles it, then restores previous interrupt
+              state and finishes executing first interrupt handler - ibex implements nonstandard MSTACK csrs to
+              save/restore state in case of nested interrupt'''
+      milestone: V2
+      test: ["irq"]
+    }
+    {
+      name: interrupt_priority_test
+      desc: '''
+            - randomly assert multiple interrupt lines at once
+            - verify that core jumps to interrupt handler corresponding to the highest priority interrupt (will
+              require multiple interrupt handler, one for each type)'''
+      milestone: V2
+      test: ["irq"]
+    }
+    {
+      name: csr_test
+      desc: '''
+            Perform all CSR instructions to implemented privileged CSR
+            Verify the reset value of the privileged CSR
+            Verify WARL field can be upated properly'''
+      milestone: V1
+      test: ["csr"]
+    }
+    {
+      name: reset_test
+      desc: '''
+            Reset the core a random number of times during program execution, core
+            should jump back to start address and re-enter the program.
+            Ensure to flush all testbench state to prevent contamination after reset.'''
+      milestone: V1
+      test: ["reset"]
+    }
+    {
+      name: perf_counter_test
+      desc: '''
+            Run random instruction test, and dump performance counters for checking'''
+      milestone: V1
+      test: ["perf"]
+    }
+    {
+      name: rv32im_instr_test
+      desc: '''
+            Specify RV32IM architecture to compiler to generate target specific
+            instructions (no compressed instructions).'''
+      milestone: V1
+      test: ["rand_instr"]
+    }
+    {
+      name: user_mode_rand_test
+      desc: '''
+            Fully randomized RV32IMC instructions, boot in U-mode,
```

Comment:
I think we need more details here. There could be many things like umode trap, security aspects of u mode verification. You can add a TODO to address it later.

Reply:
I clarified this description to specify that this test would include only random instructions, with no external stimulus or exceptions.